distributed/tests/test_worker_client.py

```diff
@@ -165,3 +168,57 @@ def mysum():
     future = c.submit(mysum)
     result = yield future._result()
     assert result == 30 * 29
+
+
+PRELOAD_TEXT = """
+_worker_info = {}
+
+def dask_setup(worker):
+    _worker_info['address'] = worker.address
+
+def get_worker_address():
+    return _worker_info['address']
+"""
+
+
+def test_worker_preload_file(loop):
+
+    def check_worker():
+        import worker_info
+        return worker_info.get_worker_address()
+
+    tmpdir = tempfile.mkdtemp()
+    path = os.path.join(tmpdir, 'worker_info.py')
+    with open(path, 'w') as f:
+        f.write(PRELOAD_TEXT)
+    with cluster(worker_kwargs={'preload': [path]}) as (s, workers),\
+            Client(s['address'], loop=loop) as c:
+
+        assert c.run(check_worker) == {
+            worker['address']: worker['address']
+            for worker in workers
+        }
+    shutil.rmtree(tmpdir)
```

Comment:
This cleanup code should be in a try-finally block so that we clean up even if the test fails.